### PR TITLE
Improve polling station form tests

### LINKED ISSUE #163: Improve polling station form tests
I'd prefer an `overrideOnce()` at the start of this test for the api response instead of relying on what's in `frontend/lib/api-mocks/PollingStationMockData.ts`.

In this case the mock plainly returns the data, so it's unlikely to lead to isses. For the `data_entries` ednpoint it does not. So I'd rather have the rule "always do an `overrideOnce()`", which leaves no doubt what data is used by test, than the rule "it depends on how complicated someone judged the mock implementation to be".

_Originally posted by @jschuurk-kr in https://github.com/kiesraad/abacus/pull/146#discussion_r1683044243_

---

The 404 response of a non-existing election id results in the same behavior as when the API returns an empty list, because of how the PollingStationProvider works. So I would like to have a test that verifies/documents this behavior.
https://github.com/kiesraad/abacus/pull/146/files#r1691007957

### PR DESCRIPTION
**Pull request by cikzh**
Closes #163 

Using `overrideOnce` at the start of every test to make clear what data is used for the test. Also making it more explicit when testing edge cases.

This PR leaves out the 404 test as discussed in the issue, as it is dependent on the decision made in #179

### CODE REVIEW

**Review by praseodym — APPROVED** [review]

**Review by jschuurk-kr — CHANGES_REQUESTED** [review]
> This PR leaves out the 404 test as discussed in the issue, as it is dependent on the decision made in https://github.com/kiesraad/abacus/issues/179

We can cover the current behavior with a test. To me, that too, is part of the zero bug policy.

File: frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx [hunk 1]
```diff
@@ -67,6 +71,7 @@ describe("Test PollingStationChoiceForm", () => {
   });
 
   test("Polling station list", async () => {
+    overrideOnce("get", "/api/polling_stations/1", 200, pollingStationMock);
```
Comment (Lionqueen94) [inline frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx:74]: Testing the list endpoint should not have an overrideOnce on the detail endpoint. It should be on "/api/elections/{election_id}/polling_stations/". See also the other instances in this file

**Review by Lionqueen94 — CHANGES_REQUESTED** [review]

File: frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx (hunk 1 above)
Comment (cikzh) [reply frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx:74]: I think I'll fix this by rebasing main and then change it. Thanks!

Comment (Lionqueen94) [inline frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx:74]:
```suggestion
    overrideOnce("get", "/api/elections/1/polling_stations", 200, pollingStationMock);
```

Comment (cikzh) [reply frontend/app/component/form/polling_station_choice/PollingStationChoiceForm.test.tsx]: Thanks, I've changed the overrides in other tests as well, since they're actually all calling the listing endpoint.

**Review by Lionqueen94 — DISMISSED** [review]

**Review by praseodym — DISMISSED** [review]

**Review by jschuurk-kr — APPROVED** [review]

**Review by Lionqueen94 — APPROVED** [review]

**Review by praseodym — APPROVED** [review]
